index.js

```diff
@@ -0,0 +1,30 @@
+const express = require('express')
+const teams = require('./teams.json')
+
+const app = express()
+
+app.get('/teams', (request, response) => {
+    response.send(teams)
+})
+
+app.get('/teams/:id', (request, response) => {
+    let abbreviationId
+    let teamId
+    abbreviationId = teams.filter((team) => {
+        return team.abbreviation === (request.params.id.toUpperCase())
+    })
+
+    teamId = teams.filter((team) => {
+        return team.id === parseInt(request.params.id)
+
+    })
```

Comment:
You're filtering twice when you don't have to. With this code regardless of what the user sends in you will loop over all teams twice making your code slow. You should instead of this distinction in your `if(IisNaN())` block so you only run the one that is necessary. Or better yet use the `OR` method described in class.